src/adapter/src/catalog.rs

```diff
@@ -5769,186 +5769,127 @@ impl Catalog {
-                        old_owner: RoleId,
-                        new_owner: RoleId,
-                    ) {
-                        let mut new_present = false;
-                        for privilege in privileges.iter_mut() {
-                            // Old owner's granted privilege are updated to be granted by the new
-                            // owner.
-                            if privilege.grantor == old_owner {
-                                privilege.grantor = new_owner;
-                            } else if privilege.grantor == new_owner {
-                                new_present = true;
-                            }
-                            // Old owner's privileges is given to the new owner.
-                            if privilege.grantee == old_owner {
-                                privilege.grantee = new_owner;
-                            } else if privilege.grantee == new_owner {
-                                new_present = true;
-                            }
-                        }
-
-                        // If the old privilege list contained references to the new owner, we may
-                        // have created duplicate entries. Here we try and consolidate them. This
-                        // is inspired by PostgreSQL's algorithm but not identical.
-                        if new_present {
-                            // Group privileges by (grantee, grantor).
-                            let privilege_map: BTreeMap<_, Vec<_>> = privileges.into_iter().fold(
-                                BTreeMap::new(),
-                                |mut accum, privilege| {
-                                    accum
-                                        .entry((privilege.grantee, privilege.grantor))
-                                        .or_default()
-                                        .push(privilege);
-                                    accum
-                                },
-                            );
-                            // Consolidate and update all privileges.
-                            *privileges = privilege_map
-                                .into_iter()
-                                .map(|((grantee, grantor), values)|
-                                    // Combine the acl_mode of all mz_aclitems with the same grantee and grantor.
-                                    values.into_iter().fold(
-                                        MzAclItem::empty(grantee, grantor),
-                                        |mut accum, mz_aclitem| {
-                                            accum.acl_mode =
-                                                accum.acl_mode.union(mz_aclitem.acl_mode);
-                                            accum
-                                        },
-                                    ))
-                                .collect();
+                Op::UpdateOwner { id, new_owner } => match id {
```

Comment:
All changes here are removing the `update_privilege_owners` nested function and whitespace changes.